Add ESLint promise checking?

We may want to add [**eslint-plugin-promise**](https://www.npmjs.com/package/eslint-plugin-promise) to the .eslintrc.js file to check for promise `.catch()` among other things. We've seen unexpected errors in other repos.

I can PR once my other in-flight ESLint PR (#85) is merged.

```diff
diff --git a/.eslintrc.js b/.eslintrc.js
index 1c40e94..50d17e4 100644
--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -8,12 +8,26 @@ module.exports = {
   "extends": [
     "eslint:recommended"
   ],
+  "plugins": [
+    "promise"
+  ],
   "root": true,
   "rules": {
+    "promise/always-return": "off",
+    "promise/no-return-wrap": "error",
+    "promise/param-names": "error",
+    "promise/catch-or-return": "error",
+    "promise/no-native": "off",
+    "promise/no-nesting": "warn",
+    "promise/no-promise-in-callback": "warn",
+    "promise/no-callback-in-promise": "warn",
+    "promise/avoid-new": "off",
+
     "eqeqeq": "error",
     "indent": ["error", 2],
     "linebreak-style": ["error", "unix"],
     "no-throw-literal": "error",
+    "no-warning-comments": "warn",
     "quotes": ["error", "double"],
     "radix": "error",
     "semi": ["error", "always"]
diff --git a/package.json b/package.json
index 441cb13..daa228e 100644
--- a/package.json
+++ b/package.json
@@ -11,6 +11,7 @@
   "devDependencies": {
     "addons-linter": "^0.15.14",
     "eslint": "^3.12.2",
+    "eslint-plugin-promise": "^3.4.0",
     "jpm": "^1.2.2",
     "npm-run-all": "^4.0.0"
   },
```

```sh
$ npm run lint:js

> testpilot-containers@0.0.1 lint:js /Users/pdehaan/dev/github/mozilla/testpilot-containers
> eslint .


/Users/pdehaan/dev/github/mozilla/testpilot-containers/index.js
  100:5  error    Expected catch() or return  promise/catch-or-return
  197:7  error    Expected catch() or return  promise/catch-or-return
  280:7  error    Expected catch() or return  promise/catch-or-return
  389:7  error    Expected catch() or return  promise/catch-or-return
  400:9  error    Expected catch() or return  promise/catch-or-return
  405:5  warning  Unexpected 'fixme' comment  no-warning-comments
  418:7  error    Expected catch() or return  promise/catch-or-return
  429:5  warning  Unexpected 'fixme' comment  no-warning-comments
  430:5  warning  Unexpected 'fixme' comment  no-warning-comments
  439:7  error    Expected catch() or return  promise/catch-or-return

/Users/pdehaan/dev/github/mozilla/testpilot-containers/webextension/js/popup.js
    8:1   warning  Unexpected 'todo' comment   no-warning-comments
   36:5   error    Expected catch() or return  promise/catch-or-return
   71:5   error    Expected catch() or return  promise/catch-or-return
  185:7   error    Expected catch() or return  promise/catch-or-return
  214:11  error    Expected catch() or return  promise/catch-or-return
  254:7   error    Expected catch() or return  promise/catch-or-return
  316:11  error    Expected catch() or return  promise/catch-or-return
  316:11  warning  Avoid nesting promises      promise/no-nesting
  411:7   error    Expected catch() or return  promise/catch-or-return
  430:5   warning  Unexpected 'fixme' comment  no-warning-comments
  449:7   error    Expected catch() or return  promise/catch-or-return

✖ 21 problems (15 errors, 6 warnings)
```

--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -8,12 +8,26 @@ module.exports = {
   "extends": [
     "eslint:recommended"
   ],
+  "plugins": [
+    "promise"
+  ],
   "root": true,
   "rules": {
+    "promise/always-return": "off",
+    "promise/avoid-new": "off",
+    "promise/catch-or-return": "error",
+    "promise/no-callback-in-promise": "warn",
+    "promise/no-native": "off",
+    "promise/no-nesting": "warn",
+    "promise/no-promise-in-callback": "warn",
+    "promise/no-return-wrap": "error",
+    "promise/param-names": "error",
+
     "eqeqeq": "error",
     "indent": ["error", 2],
     "linebreak-style": ["error", "unix"],
     "no-throw-literal": "error",
+    "no-warning-comments": "warn",
     "quotes": ["error", "double"],
     "radix": "error",
     "semi": ["error", "always"]

--- a/package.json
+++ b/package.json
@@ -11,6 +11,7 @@
   "devDependencies": {
     "addons-linter": "^0.15.14",
     "eslint": "^3.12.2",
+    "eslint-plugin-promise": "^3.4.0",
     "jpm": "^1.2.2",
     "npm-run-all": "^4.0.0"
   },

